### PR TITLE
[ch3751] Fix wording in my donation page

### PR DESCRIPTION
**Pull request by cacahouwete**
https://app.clubhouse.io/enmarche/story/3751/emp%C3%AAcher-les-personnes-qui-font-d%C3%A9j%C3%A0-un-don-mensuel-d-en-faire-un-second

### CODE REVIEW

File: src/Repository/DonationRepository.php [hunk 1]
```diff
@@ -52,4 +52,21 @@ public function findAllSubscribedDonationByEmail(string $email): array
...
+    {
+        return $this->createQueryBuilder('donation')
+            ->andWhere('donation.emailAddress = :email')
+            ->andWhere('donation.duration != :duration')
+            ->andWhere('donation.subscriptionEndedAt IS NOT NULL')
+            ->setParameters([
+                'email' => $email,
+                'duration' => PayboxPaymentSubscription::NONE,
+            ])
+            ->orderBy('donation.donatedAt', 'ASC')
```
Comment (B-Galati) [inline src/Repository/DonationRepository.php]: It should be DESC instead no ?

Comment (B-Galati) [reply src/Repository/DonationRepository.php]: Do we need to check the status of the Donation ?

File: src/Controller/EnMarche/UserController.php [hunk 2]
```diff
@@ -46,6 +46,7 @@ public function profileDonationAction(DonationRepository $donationRepository): R
         return $this->render('user/my_donation.html.twig', [
             'donations' => $donationRepository->findByEmailAddressOrderedByDonatedAt($userEmail, 'DESC'),
             'subscribed_donations' => $donationRepository->findAllSubscribedDonationByEmail($userEmail),
+            'last_cancel_subscription' => $donationRepository->findLastSubscribedDonationByEmail($userEmail),
```
Comment (B-Galati) [inline src/Controller/EnMarche/UserController.php]: I really don't get the semantic here..

We look for last canceled subscription but repository look for last subscribed donation ?! wtf ?

File: src/Repository/DonationRepository.php [hunk 3]
```diff
@@ -47,23 +47,22 @@ public function findAllSubscribedDonationByEmail(string $email): array
...
         return $this->createQueryBuilder('donation')
             ->andWhere('donation.emailAddress = :email')
-            ->andWhere('donation.duration != :duration')
             ->andWhere('donation.subscriptionEndedAt IS NOT NULL')
             ->setParameters([
                 'email' => $email,
                 'duration' => PayboxPaymentSubscription::NONE,
             ])
-            ->orderBy('donation.donatedAt', 'ASC')
+            ->orderBy('donation.donatedAt', 'DESC')
```
Comment (B-Galati) [inline src/Repository/DonationRepository.php]: It should be ordered by subscriptionEndedAt no ?

Comment (cacahouwete) [reply src/Repository/DonationRepository.php]: Oooo yes, my brain not started yet ^^,

File: templates/user/my_donation.html.twig [hunk 4]
```diff
@@ -15,18 +15,20 @@
...
                 </a>. <br />
                 {% for subscribed_donation in subscribed_donations %}
-                    Prochaine échéance du don mensuel de {{ subscribed_donation.amountInEuros }}€ le {{ subscribed_donation.nextDonationAt|date('d M Y') }}
+                    Prochaine échéance du don mensuel de {{ subscribed_donation.amountInEuros }}€ le {{ subscribed_donation.nextDonationAt|localizeddate('medium', 'none') }}
                     {% if not loop.last %}
                         <br />
                     {% endif %}
                 {% endfor %}
+            {% elseif last_subscription_ended %}
+                Vous avez mis fin à votre don mensuel le {{ last_cancel_subscription.subscriptionEndedAt|localizeddate('medium', 'none') }}. Vous pouvez en refaire un <a href="{{ path('donation_index') }}">en cliquant ici</a>.
```
Comment (illusionOfParadise) [inline templates/user/my_donation.html.twig]: This variable does not exists. It seems that after changing this line https://github.com/EnMarche/en-marche.fr/pull/2409/files#diff-7afab20efac876cf49719e435cc8f688R49 you forget to change variable name.

Comment (cacahouwete) [reply templates/user/my_donation.html.twig]: My brain is totaly disconnected from earth this morning ^^

**Comment by ottaviano** [comment]
don't forget to test your dev 😈

File: templates/user/my_donation.html.twig (hunk 4 above)
Comment (B-Galati) [reply templates/user/my_donation.html.twig]: Stop finding yourself excuses OMG

**Pull request merged**